### PR TITLE
Enhance UI for readonly view picker

### PR DESCRIPTION
**Pull request by zhyuanqi**
### Description

Enhance UI for readonly view picker

## Screenshot

Screenshot 2024-04-10 at 12 09 21 AM

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx [hunk 1]
```diff
@@ -5,22 +5,24 @@
...
 import { DataSourceOption } from '../data_source_menu/types';
 import { getDataSourceById } from '../utils';
-import { MenuPanelItem } from '../../types';
 
 interface DataSourceViewProps {
   fullWidth: boolean;
   selectedOption: DataSourceOption[];
-  savedObjectsClient?: SavedObjectsClientContract;
-  notifications?: ToastsStart;
+  savedObjectsClient: SavedObjectsClientContract;
```
Comment (zhyuanqi) [inline src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx:17]: SavedObjctClient and notification should be required according to this doc: https://quip-amazon.com/UepCAPYFGtKC/Use-Data-Source-Menu-to-Nav-Bar-or-Data-Source-Selector-inside-Page

**Comment by BionIT** [comment]
Can we add more context of the PR? In the `Testing the changes` section, can we add steps that were performed for the tests?

File: src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx (hunk 1 above)
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx:17]: savedobject client and notification are not required for this component, can you change it back to optional?

File: src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx [hunk 2]
```diff
@@ -78,67 +85,44 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
...
-          id={'dataSourceViewContextMenuPopover'}
-          button={button}
-          isOpen={this.state.isPopoverOpen}
-          closePopover={this.closePopover.bind(this)}
-          panelPaddingSize="none"
-          anchorPosition="downLeft"
+      <div>
+        <EuiToolTip
+          data-test-subj="dataSourceViewrToolTip"
+          title={<div style={{ width: '300px' }}>SELECTED DATA SOURCE</div>}
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx:101]: can we move style into a css file? Does the width break mobile view?

File: src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx [hunk 3]
```diff
@@ -45,31 +48,35 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
...
+        this.handleDataSourceFetchError(optionId, error);
       }
     }
   }
 
+  handleDataSourceFetchError(optionId: string, error?) {
+    this.props.notifications.addWarning(
+      i18n.translate('dataSource.fetchDataSourceError', {
+        defaultMessage: error
+          ? `Failed to fetch data source due to ${error}`
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx:74]: Noticed that the error messages we display is getting diverged among components, can we extract datasourceFetchError as util function to be used by all components?

File: src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx (hunk 1 above)
Comment (zhyuanqi) [reply src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx]: got it

File: src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx (hunk 2 above)
Comment (zhyuanqi) [reply src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx]: Would do

File: src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx (hunk 3 above)
Comment (zhyuanqi) [reply src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx]: okay

File: src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx [hunk 4]
```diff
@@ -45,29 +68,38 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
 
     const option = selectedOption[0];
     const optionId = option.id;
+    const defaultDataSource = this.props.uiSettings?.get('defaultDataSource', null) ?? null;
+    const allDataSources = await getDataSources(this.props.savedObjectsClient!);
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx:72]: why do we need to get all data sources?

File: src/plugins/data_source_management/public/components/data_source_view/_data_source_view.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.panelWidth {
```
[the remark this answers is not in the document]
Comment (seraphjiang) [reply src/plugins/data_source_management/public/components/data_source_view/_data_source_view.scss:1]: Restrict scope to the component only

File: src/plugins/data_source_management/public/components/data_source_view/_data_source_view.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
...
+  width: 300px;
+}
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_view/_data_source_view.scss:3]: Let's follow guideline for style file and class name https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/8667700544/job/23771246546?pr=6416

**Comment by zhyuanqi** [comment]
Close this pull request as new UI requirement has given

**Pull request closed without merging**